### PR TITLE
fix: preserve client-provided graph_id and auto-populate thread_name

### LINKED ISSUE #231: [BUG] thread_name in /threads/search return empty string
**Describe the bug**
thread_name in `/threads/search` return empty string

**To Reproduce**
Input:
```bash
curl 'http://127.0.0.1:20024/threads/search' \
  -H 'Accept: */*' \
  -H 'content-type: application/json' \
  --data-raw '{"limit":20,"offset":0,"sort_by":"updated_at","sort_order":"desc"}'
```

Image

Output:
```json
[
    {
        "thread_id": "615ec1e9-f33e-4b2e-bd1f-84826b82a6e2",
        "status": "idle",
        "metadata": {
            "owner": "anonymous",
            "graph_id": "xxx",
            "thread_name": "",
            "assistant_id": "083660f4-e80d-54ad-bd52-c617093e1d12"
        },
        "user_id": "anonymous",
        "created_at": "2026-03-07T17:27:10.751930Z",
        "updated_at": "2026-03-07T17:27:11.728948Z"
    },
    {
        "thread_id": "2885b549-ab58-42d1-8af9-82323a7997ce",
        "status": "idle",
        "metadata": {
            "owner": "anonymous",
            "graph_id": "xxx",
            "thread_name": "",
            "assistant_id": "083660f4-e80d-54ad-bd52-c617093e1d12"
        },
        "user_id": "anonymous",
        "created_at": "2026-03-07T17:26:47.241269Z",
        "updated_at": "2026-03-07T17:26:56.719907Z"
    },
    {
        "thread_id": "1031a54c-bcff-46c6-9f0a-91d69bdb743e",
        "status": "idle",
        "metadata": {
            "owner": "anonymous",
            "graph_id": "xxx",
            "thread_name": "",
            "assistant_id": "083660f4-e80d-54ad-bd52-c617093e1d12"
        },
        "user_id": "anonymous",
        "created_at": "2026-03-07T17:19:21.523190Z",
        "updated_at": "2026-03-07T17:22:38.978523Z"
    }
]
```

Image

Deepagents UI can not show thread name

Image

**Expected behavior**
LangChain Agent Server could return thread name

**Actual behavior**
Aegra could also return thread name

**Environment:**
- OS: Docker Image `python:3.13.12`
- Python version: 3.13.12
- Aegra version: 0.7.5
- LangGraph version: 1.0.10

**Configuration:**
- Database: PostgreSQL 18
- Auth type: noop
- Deployment: Docker

**Logs/Error Messages:**
```
026-03-07T17:27:28.799901Z [info     ] 172.18.0.1:38848 - "POST /threads/search HTTP/1.1" 200 [app.access_logs] duration=10152931 filename=logger_middleware.py func_name=__call__ http={'url': '/threads/search', 'status_code': 200, 'method': 'POST', 'version': '1.1', 'request_id': '0a46e8d7b6cb477bba7383f3f8692bfe'} lineno=90 network={'client': {'ip': '172.18.0.1', 'port': 38848}}

2026-03-07T17:27:34.933376Z [info     ] 172.18.0.1:36332 - "POST /threads/search HTTP/1.1" 200 [app.access_logs] duration=6984817 filename=logger_middleware.py func_name=__call__ http={'url': '/threads/search', 'status_code': 200, 'method': 'POST', 'version': '1.1', 'request_id': '5cc06940dcbf42138f705d3c520d1686'} lineno=90 network={'client': {'ip': '172.18.0.1', 'port': 36332}}

2026-03-07T17:27:38.460613Z [info     ] 172.18.0.1:36332 - "POST /threads/search HTTP/1.1" 200 [app.access_logs] duration=6232081 filename=logger_middleware.py func_name=__call__ http={'url': '/threads/search', 'status_code': 200, 'method': 'POST', 'version': '1.1', 'request_id': '24588dff2f18496fa5ed8c7028b5b1f6'} lineno=90 network={'client': {'ip': '172.18.0.1', 'port': 36332}}

2026-03-07T17:27:38.483318Z [info     ] 172.18.0.1:36332 - "POST /threads/615ec1e9-f33e-4b2e-bd1f-84826b82a6e2/history HTTP/1.1" 200 [app.access_logs] duration=19794522 filename=logger_middleware.py func_name=__call__ http={'url': '/threads/615ec1e9-f33e-4b2e-bd1f-84826b82a6e2/history', 'status_code': 200, 'method': 'POST', 'version': '1.1', 'request_id': '0188b1a6bfc64105bb93f0cc834ae640'} lineno=90 network={'client': {'ip': '172.18.0.1', 'port': 36332}}

2026-03-07T17:27:38.483849Z [info     ] 172.18.0.1:36340 - "POST /assistants/search HTTP/1.1" 200 [app.access_logs] duration=15467371 filename=logger_middleware.py func_name=__call__ http={'url': '/assistants/search', 'status_code': 200, 'method': 'POST', 'version': '1.1', 'request_id': '3e0d8678a65b462aa80f992db5c5771a'} lineno=90 network={'client': {'ip': '172.18.0.1', 'port': 36340}}

2026-03-07T17:28:02.735387Z [info     ] 172.18.0.1:32924 - "POST /threads/search HTTP/1.1" 200 [app.access_logs] duration=8134179 filename=logger_middleware.py func_name=__call__ http={'url': '/threads/search', 'status_code': 200, 'method': 'POST', 'version': '1.1', 'request_id': '3ed063d1d4774eb1bf3b1d4bc7265060'} lineno=90 network={'client': {'ip': '172.18.0.1', 'port': 32924}}

2026-03-07T17:28:27.889411Z [info     ] 172.18.0.1:56052 - "POST /threads/search HTTP/1.1" 200 [app.access_logs] duration=7427464 filename=logger_middleware.py func_name=__call__ http={'url': '/threads/search', 'status_code': 200, 'method': 'POST', 'version': '1.1', 'request_id': '57e4dd7a78b745cd9c548fb3e96ffcb4'} lineno=90 network={'client': {'ip': '172.18.0.1', 'port': 56052}}

2026-03-07T17:32:41.659748Z [info     ] 172.18.0.1:40810 - "POST /threads/search HTTP/1.1" 200 [app.access_logs] duration=10553511 filename=logger_middleware.py func_name=__call__ http={'url': '/threads/search', 'status_code': 200, 'method': 'POST', 'version': '1.1', 'request_id': 'b65e35430cd44adabe209ef5bb92e820'} lineno=90 network={'client': {'ip': '172.18.0.1', 'port': 40810}}

2026-03-07T17:32:50.940634Z [info     ] 172.18.0.1:38782 - "POST /threads/search HTTP/1.1" 200 [app.access_logs] duration=8010216 filename=logger_middleware.py func_name=__call__ http={'url': '/threads/search', 'status_code': 200, 'method': 'POST', 'version': '1.1', 'request_id': '14f628365f654fb594844a3852ca2eb3'} lineno=90 network={'client': {'ip': '172.18.0.1', 'port': 38782}}
```

### LINKED ISSUE #254: [BUG] create_thread in aegra always overwrites graph_id to None in thread metadata
Describe the bug
                                                                       
  create_thread in aegra always overwrites graph_id to None in thread
  metadata, ignoring any graphId passed by the client. This causes     
  update_thread_state to fail with "Thread has no associated graph. 
  Cannot update state."                                                
                                                                  
  To Reproduce                                                         
   
  1. Create a thread with graphId specified:                           
```typescript
  const client = new Client({ apiKey: token, apiUrl: BASE_AGENT_URL });
                                                                       
  await client.threads.create({                                        
    threadId: 'test-thread-123',                                       
    metadata: { graphId: 'agent' },                                    
    graphId: 'agent',                                                  
    ifExists: 'do_nothing',                                            
  });                                                  
```                
                                                                       
  2. Attempt to update thread state:                                   
```typescript
  await client.threads.updateState('test-thread-123', {                
    values: { messages: { id: 'msg-1', type: 'system', content: 'test'
  } },                                                                 
    asNode: 'agent',                                                   
  });      
```                                                            
                                                                       
  3. See error.                                                   
                                                                       
  Expected behavior                                               
                                                                       
  The thread should store the provided graph_id and update_thread_state
   should succeed.
                                                                       
  Actual behavior                                                      
   
  update_thread_state returns a 400 error:                             
```
  {"error":"bad_request","message":"Thread                        
  '8b2f74ca-eabc-4c62-9c7b-8fefd8e4c478' has no associated graph.      
  Cannot update state.","details":null}               
```                 
                                       
  Environment:                                                         
  - Aegra (version from python-agent/.venv)                            
  - LangGraph SDK client (JS)                                          
                                                                       
  Configuration:                                                       
  - Deployment: local / docker                                    
                                                                       
  Logs/Error Messages:                                            
```
  {"error":"bad_request","message":"Thread                             
  '8b2f74ca-eabc-4c62-9c7b-8fefd8e4c478' has no associated graph. 
  Cannot update state.","details":null}             
```                   
                                       
  Root Cause                                                           
                                                                       
  In threads.py:159-167, create_thread unconditionally sets graph_id:  
  None:                                                                
                                                                       
```python
  metadata.update(                                                     
      {                                                                
          "owner": user.identity,                                      
          "assistant_id": None,                                        
          "graph_id": None,          # <-- always overwrites to None   
          "thread_name": mget("thread_name", ""),                      
      }                                                                
  )              
```                                                      
                                                                       
  Then update_thread_state (line 416-421) checks this value and raises 
  a 400 when it's falsy.                                          
                                                                       
  Possible Solution                                                    
   
  Preserve graph_id from request metadata instead of overwriting it:   
                                                                  
```python
  metadata.update(                                                     
      {                                                                
          "owner": user.identity,
          "assistant_id": None,                                        
          "graph_id": metadata.get("graph_id") or                      
  metadata.get("graphId"),                                             
          "thread_name": metadata.get("thread_name", ""),              
      }                                                                
  )               
```                                                
                                                                       
  Also note the casing mismatch: the JS client sends graphId           
  (camelCase) but aegra stores graph_id (snake_case) — the fix should
  handle both.

### PR DESCRIPTION
**Pull request by ibbybuilds**
## Summary
- **Fixes #254**: `create_thread` no longer overwrites `graph_id` to `None` — uses `setdefault` so client-provided values are preserved. Normalizes JS SDK camelCase `graphId` to `graph_id`.
- **Fixes #231**: `thread_name` is now auto-populated from the first message content when a run starts, so UIs (deepagents, agent-chat-ui) can show readable thread titles instead of empty strings.
- Bumps version to 0.8.4

## Test plan
- [x] 3 new integration tests for graph_id preservation (snake_case, camelCase, precedence)
- [x] 11 new unit tests for `_extract_thread_name` (edge cases: empty input, non-string content, truncation, langchain objects)
- [x] All 68 existing thread + run tests pass with no regressions
- [ ] E2E verification against running server

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Thread names auto-derived from the first human message and truncated when long.
  * Thread creation now preserves client-provided metadata and always records the authenticated user as owner; client camelCase keys are preserved rather than normalized.

* **Tests**
  * Added unit and integration tests for thread-name extraction, truncation, role/type filtering, and metadata handling.

* **Chores**
  * Version bumped to 0.8.4 for aegra-api and aegra-cli.

### CODE REVIEW

File: libs/aegra-api/src/aegra_api/api/runs.py [hunk 1]
```diff
@@ -111,20 +111,52 @@ async def set_thread_status(session: AsyncSession, thread_id: str, status: str)
...
+    return ""
+
+
 async def update_thread_metadata(
     session: AsyncSession,
     thread_id: str,
     assistant_id: str,
     graph_id: str,
     user_id: str | None = None,
+    input_data: dict[str, Any] | None = None,
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/src/aegra_api/api/runs.py:182]: Fixed in 90003f8. Added `*` separator before `user_id` and `input_data` params.

File: libs/aegra-api/src/aegra_api/api/runs.py [hunk 2]
```diff
@@ -111,20 +111,52 @@ async def set_thread_status(session: AsyncSession, thread_id: str, status: str)
...
+        content: str | None = None
+        if isinstance(msg, dict):
+            content = msg.get("content")
+        elif hasattr(msg, "content"):
+            content = getattr(msg, "content", None)
+        if isinstance(content, str) and content.strip():
+            name = content.strip()
+            if len(name) > _THREAD_NAME_MAX_LENGTH:
+                return name[:_THREAD_NAME_MAX_LENGTH].rsplit(" ", 1)[0] + "..."
+            return name
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/src/aegra_api/api/runs.py:171]: Fixed in 90003f8. Added _resolve_content_text() that handles both plain strings and list-of-blocks format. Added 3 tests covering list blocks, multiple text blocks, and image-only blocks fallback.

File: libs/aegra-api/tests/unit/test_api/test_runs_thread_name.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                ]
+            }
+        )
+        assert result == "Real content"
+
+    def test_truncates_long_content_at_word_boundary(self) -> None:
+        long_content = "word " * 30  # 150 chars
+        result = _extract_thread_name({"messages": [{"content": long_content}]})
+        assert result.endswith("...")
+        assert len(result) <= 104  # 100 + "..."
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/tests/unit/test_api/test_runs_thread_name.py]: Fixed in 90003f8. Updated comment to reflect actual length (148 chars after strip).

File: libs/aegra-api/tests/unit/test_api/test_runs_thread_name.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        result = _extract_thread_name(
+            {"messages": [{"role": "human", "content": None}]}
+        )
+        assert result == ""
+
+    def test_returns_empty_for_non_string_content(self) -> None:
+        result = _extract_thread_name(
+            {"messages": [{"role": "human", "content": 42}]}
+        )
+        assert result == ""
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/tests/unit/test_api/test_runs_thread_name.py]: Fixed in 90003f8. Ran ruff format on both files.

File: libs/aegra-api/src/aegra_api/api/runs.py [hunk 5]
```diff
@@ -242,7 +300,9 @@ async def create_run(
 
     # Mark thread as busy and update metadata with assistant/graph info
     # update_thread_metadata will auto-create thread if it doesn't exist
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/src/aegra_api/api/runs.py:314]: Fixed in c54c762. All three call sites now use user_id=user.identity as keyword arg.

File: libs/aegra-api/src/aegra_api/api/runs.py [hunk 6]
```diff
@@ -111,20 +111,75 @@ async def set_thread_status(session: AsyncSession, thread_id: str, status: str)
...
+        if isinstance(msg, dict):
+            raw_content = msg.get("content")
+        elif hasattr(msg, "content"):
+            raw_content = getattr(msg, "content", None)
+        text = _resolve_content_text(raw_content)
+        if text.strip():
+            name = text.strip()
+            if len(name) > _THREAD_NAME_MAX_LENGTH:
+                return name[:_THREAD_NAME_MAX_LENGTH].rsplit(" ", 1)[0] + "..."
+            return name
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/src/aegra_api/api/runs.py:171]: Fixed in c54c762. Now filters to human/user role only, skips system/assistant messages. Added tests for role filtering and OpenAI-style user role.

File: libs/aegra-api/src/aegra_api/api/runs.py [hunk 7]
```diff
@@ -242,7 +300,9 @@ async def create_run(
 
     # Mark thread as busy and update metadata with assistant/graph info
     # update_thread_metadata will auto-create thread if it doesn't exist
-    await update_thread_metadata(session, thread_id, assistant.assistant_id, assistant.graph_id, user.identity)
+    await update_thread_metadata(
+        session, thread_id, assistant.assistant_id, assistant.graph_id, user.identity, input_data=request.input
+    )
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/src/aegra_api/api/runs.py:317]: Fixed in c54c762. All three call sites now pass user_id as keyword arg.
